scripts/templates/pawseyenv.lua

```diff
@@ -11,6 +11,30 @@
 
 ]]--
 
+--[[
+-- There has been issues where the module just fails on first assert and it is unclear
+-- from the error message what that issue is. Thus the module also checks for the 
+-- definition of HOME, USER, PAWSEY_PROJECT etc. 
+--]]
+
+local function check_envs() 
+  local envvars = {"HOME", "USER", "PAWSEY_PROJECT"}
+  for ind,var in ipairs(envvars) do 
+    local env = os.getenv(var)
+    if env == nil or env == "" then 
+      nid = os.getenv("HOSTNAME")
```

Comment:
what happens if `HOSTNAME` itself is not set? -- as unlikely as HOME/USER I reckon.

maybe you could substitute with a Lua command that executes the `hostname` command